library/core/src/ptr/mut_ptr.rs

```diff
@@ -100,6 +100,36 @@ impl<T: ?Sized> *mut T {
         if self.is_null() { None } else { unsafe { Some(&*self) } }
     }
 
+    /// Returns `None` if the pointer is null, or else returns a reference to
+    /// the value wrapped in `Some`.
+    ///
+    /// # Safety
+    ///
+    /// While this method and its mutable counterpart are useful for
```

Comment:
Above you removed the "and its counterpart", I think it would make sense to also remove it here (and in `as_ref`). Alternatively, link to what that counterpart is.